### PR TITLE
Electron app enhancements

### PR DESCRIPTION
**Pull request by Shubhdeep12**
Before opening this PR:
* [x] I added a [Changeset Entry](https://spotlightjs.com/contribute/changesets/) with `pnpm changeset:add`
* [x] I referenced issues that this PR addresses
#302 
#304

### CODE REVIEW

**Comment by HazAT** [comment]
Pretty cool, but I added the error dialog already, it should be in the code https://github.com/getsentry/spotlight/blob/main/packages/electron/src/electron/main/index.ts#L224

**Comment by Shubhdeep12** [comment]
> Pretty cool, but I added the error dialog already, it should be in the code https://github.com/getsentry/spotlight/blob/main/packages/electron/src/electron/main/index.ts#L224
> 
> Forgot to check/close the box

yeah, I missed it. it seems like we will be asking the user for the first time until sentry.('sentry-enabled') is undefined.
i removed the processes to capture exceptions and added the function to show error page whenever Sentry is about to send data ( in askForPermissionToSendToSentry function)

cc @HazAT

File: packages/electron/index.html [hunk 1]
```diff
@@ -11,9 +11,43 @@
...
     </style>
   </head>
   <body>
+    <div id="error-screen" style="display: none">
+      <div class="error-page-navbar">
+        <img alt="spotlight-icon" src="./resources/sized.png" width="50" height="50" />
+        <p class="spotlight-title">Spotlight</p>
+      </div>
+      <h1 class="error header">Oops! An error occurred.</h1>
+      <p class="error description">Press Ctrl/Cmd + R to reload the app.</p>
```
Comment (HazAT) [inline packages/electron/index.html:48]:
```suggestion
      <p class="error description">Press Cmd + R to reload the app.</p>
```

**Review by HazAT — APPROVED** [review]
👍

**Pull request merged**